pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -1,21 +1,46 @@
 import pytest
 
 import numpy as np
-from datetime import date, timedelta, time
+from datetime import date, timedelta, time, datetime
 
 import dateutil
 import pandas as pd
 import pandas.util.testing as tm
 from pandas.compat import lrange
 from pandas.compat.numpy import np_datetime64_compat
 from pandas import (DatetimeIndex, Index, date_range, DataFrame,
-                    Timestamp, datetime, offsets)
+                    Timestamp, offsets)
 
 from pandas.util.testing import assert_almost_equal
 
 randn = np.random.randn
 
 
+class TestDatetimeIndexLikeTimestamp(object):
+    # Tests for DatetimeIndex behaving like a vectorized Timestamp
```

Comment:
I really like the vectorized-timestamp abstraction.  It might make sense to have parallel test classes for Timestamp methods and their DatetimeIndex counterparts.